apps/src/music/blockly/MusicBlocklyWorkspace.ts

```diff
@@ -192,26 +192,27 @@ export default class MusicBlocklyWorkspace {
    * @param blockMode Current block mode, such as "simple2" or "advanced"
    */
   compileSong(scope: object, blockMode: ValueOf<typeof BlockMode>) {
-    if (!this.workspace) {
+    const workspace = this.workspace;
```

Comment:
TypeScript was complaining throughout this file because `this.workspace` could be null in each place where it is accessed, even with this early return. This is because it's not capable of knowing whether the workspace value on the class instance could have changed out from under us. To reassure it that we're talking about the same workspace through, I created a local constant.